packages/motherduck/test/Adaptor.test.js

```diff
@@ -0,0 +1,885 @@
+        database: 'test_db',
+        testMode: true // Enable test mode to use mock connection
+      },
+      data: null, // OpenFN standard is to start with null data
+      references: [] // No queries array - this is adapter-specific
+    };
+  });
+
+  describe('execute', () => {
+    it('should execute a sequence of operations', async () => {
+      const operation = execute(
+        query('SELECT 1 as test_value'),
+        (state) => {
+          expect(state.data).to.exist;
+          expect(state.data).to.be.an('array');
+          return state;
+        }
+      );
+      
+      const result = await operation(baseState);
+      expect(result.data).to.exist;
+      expect(result.references).to.be.an('array');
+    });
+
+    it('should preserve state.data between operations', async () => {
+      const operation = execute(
+        (state) => ({ ...state, data: { step: 1 } }),
+        (state) => {
+          expect(state.data.step).to.equal(1);
+          return { ...state, data: { ...state.data, step: 2 } };
+        }
+      );
+      
+      const result = await operation(baseState);
+      expect(result.data.step).to.equal(2);
+    });
+
+    it('should handle errors in operation chain', async () => {
+      const operation = execute(
+        (state) => { throw new Error('Test error'); }
+      );
+      
+      try {
+        await operation(baseState);
+        expect.fail('Should have thrown an error');
+      } catch (error) {
+        expect(error.message).to.equal('Test error');
+      }
+    });
+  });
+
+  describe('query', () => {
+    it('should execute a simple SELECT query', async () => {
+      const operation = execute(
+        query('SELECT 1 as test_number, \'hello\' as test_string')
+      );
+      const result = await operation(baseState);
+      
+      expect(result.data).to.be.an('array');
+      expect(result.response).to.exist;
+      expect(result.response.rowCount).to.be.a('number');
+    });
+
+    it('should handle complex analytical queries', async () => {
+      const operation = execute(
+        query(`
+          WITH test_data AS (
+            SELECT 1 as id, 'Test' as name
+            UNION ALL
+            SELECT 2 as id, 'Data' as name
+          )
+          SELECT COUNT(*) as total_records FROM test_data
+        `)
+      );
+      
+      const result = await operation(baseState);
+      expect(result.data).to.be.an('array');
+      expect(result.response.command).to.equal('WITH');
+    });
+
+    it('should expand references in query text', async () => {
+      const tableName = 'test_users_ref';
+      const minAge = 18;
+      
+      const operation = execute(
+        createTable(tableName, [{ name: 'id', type: 'INTEGER' }, { name: 'age', type: 'INTEGER' }]),
+        insert(tableName, { id: 1, age: 25 }),
+        query(`SELECT * FROM ${tableName} WHERE age >= ${minAge}`)
+      );
+      
+      const result = await operation(baseState);
+      expect(result.data).to.exist;
```

Comment:
Another little problem we have in this test is that we don't know whether the expand reference actually worked. If you comment out the `expandReferences` call from `createTable` the test will probably still pass